### ISSUE TITLE
Pyright parser uses wrong order of operations for binary operations

### ISSUE THREAD

**Issue by erictraut** [python/mypy#10243]
When binary operators are used in chains, pyright doesn't use the correct order of operations in some cases. See below for an example of a false positive error due to this issue.

```python
v1 = "1" in "1" == "1"  # Incorrectly generates error
v2 = ("1" in "1") == "1"
v3 = "1" in ("1" == "1") # Correctly generates error
```

**Comment by erictraut** [comment]
Sorry, entered this bug into the wrong repo. Closing.
